### PR TITLE
Using drop_overload category in EDS to report drop_overload stats in LRS

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
Using drop_overload category in EDS to report drop_overload stats in LRS.

This is a follow up PR to support drop_overload load report service: https://github.com/envoyproxy/envoy/pull/31384

Currently it is reporting with a fixed category "drop_overload".  This PR changes it into the category passed in by cluster or EDS policy configuration.

### CODE REVIEW

File: source/common/upstream/cluster_manager_impl.cc
Comment (adisuissa) [inline source/common/upstream/cluster_manager_impl.cc:1250]: nit: move the drop_category.

File: source/server/admin/config_dump_handler.cc
Comment (adisuissa) [inline source/server/admin/config_dump_handler.cc:314]: Seems that the [drop_percentage](https://github.com/envoyproxy/envoy/blob/b10deb602383972cd71ab0a836c846f7eb91b790/api/envoy/config/endpoint/v3/endpoint.proto#L51) in the input is of type `FractionalPercent` (which is the same as the config-dump.
IMHO it makes more sense to keep the exported value the same (so one can easily compare the config-dump to the input).

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
Please add a release note as this is a user-facing change.
/assign-from @envoyproxy/senior-maintainers

File: source/server/admin/config_dump_handler.cc
Comment (yanjunxiang-google) [reply source/server/admin/config_dump_handler.cc:314]: Is your question that if input is numerator : 20, denominator: HUNDRED, the config dump converted that into MILLION based?

Comment (adisuissa) [reply source/server/admin/config_dump_handler.cc:314]: Yes. It seems that it overrides the input. I understand that it will also update the numerator, but it just seems unintuitive from a user's perspective. BTW this is not a blocker for this PR, just a comment for a future improvement.

Comment (yanjunxiang-google) [reply source/server/admin/config_dump_handler.cc:314]: BTW, the config dump choose MILLION based, is to cover the case that the drop_ratio is very small, like 0.00001.

**Review by mattklein123 — COMMENTED** [review]
LGTM please merge main.

/wait

File: test/mocks/upstream/cluster.h
Comment (yanjunxiang-google) [reply test/mocks/upstream/cluster.h:37]: I need to add this drop_category_ back as it is needed for the setDropCategory() mock function. Without it, the CI is failing.

**Comment by yanjunxiang-google** [comment]
Kind Ping

File: test/mocks/upstream/cluster.h
Comment (adisuissa) [reply test/mocks/upstream/cluster.h:37]: I don't think this is required, as the default ON_CALL can return the same string ("drop_overload"), but I'm not going to block this PR.

File: changelogs/current.yaml
Comment (adisuissa) [inline changelogs/current.yaml:134]:
```suggestion
    It is changed to use drop_category that is set in the ClusterLoadAssignment.Policy.DropOverload.category
```

EDS is not really necessary here.
Please also add a link to the field of `ClusterLoadAssignment.Policy.DropOverload.category`.

File: test/mocks/upstream/cluster.h
Comment (yanjunxiang-google) [reply test/mocks/upstream/cluster.h:37]: Without drop_category_, how do we define the mock function setDropCategory() in cluster.cc ?

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**
